src/NSrtm.Core/BiCubicInterpolation/BivariatePolynomial.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+
+namespace NSrtm.Core.BicubicInterpolation
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1815:OverrideEqualsAndOperatorEqualsOnValueTypes", Justification = "This value type would never be compared")]
+    public struct BivariatePolynomial
+    {
+        private readonly double[][] _coefficients;
+
+        public BivariatePolynomial([NotNull] IReadOnlyList<double> coefficients)
+        {
+            if (coefficients == null) throw new ArgumentNullException("coefficients");
+            if (coefficients.Count != 16) throw new ArgumentException("Bivariate polynomial use 16 coefficients.", "coefficients");
+            _coefficients = coefficients.Select((c, i) => new {Index = i, value = c})
```

Comment:
I don't store coeff in IROL, i only pass them to constructor in this form. Inside i use array. I think passing readonly collection to public methods is a good habit, am i wrong?